tools/login/actions/login.php

```diff
@@ -113,8 +126,17 @@
 if ($_REQUEST["action"] == "logout") {
     $this->LogoutUser();
     $this->SetMessage(_t('LOGIN_YOU_ARE_NOW_DISCONNECTED'));
-    $this->Redirect(preg_replace('/(&|\\\?)$/m', '', preg_replace('/(&|\\\?)action=logout(&)?/', '$1', $incomingurl)));
-    $this->exit();
+    
+    if (isset($loggedouturl))
+    {
+    	$this->Redirect(preg_replace('/(&|\\\?)$/m', '', preg_replace('/(&|\\\?)action=logout(&)?/', '$1', $incomingurl)));
+    	$this->exit();
+    }
+    else
+    {
+    	$this->Redirect($this->href('', $loggedouturl, ''));   
+    	$this->exit();
+    }
```

Comment:
```suggestion
    } else {
      $this->Redirect($loggedouturl);
    }
```
pour avoir la compatibilité avec les urls complètes

Reply:
J'ai fait ce choix car je comprenais qu'il pouvait y avoir des transferts des anciens parametres de l'url appelante dans la nouvelle si on ne le faisait pas... Ici je repartais de 0. J'ai eu du mal à comprendre les parametres de Redirect.

Reply:
les paramètres de `redirect` sont assez simples : https://github.com/YesWiki/yeswiki/blob/862b4e1809b2b04129012c0d0d40a52d425f6d41/includes/YesWiki.php#L402
C'est plutôt `YesWiki->Href` qui est un peut plus complexe.
Et si `$loggedouturl` n'est pas une page alors l'appel de `YesWiki->Href` va mal se passer.

peut-être que tu voulais appeler
`$this->generateLink($urlOrPageWithParams)` https://github.com/YesWiki/yeswiki/blob/862b4e1809b2b04129012c0d0d40a52d425f6d41/includes/YesWiki.php#L540
qui permet de générer le bon lien en vérifiant si c'est une url, une page une page avec des paramètres.

et ce serait plus simple de faire
```php
$this->Redirect(empty($loggedouturl) ? $this->Href() : $this->generateLink($loggedouturl));
```
Qu'en dis-tu ?